src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -531,15 +532,19 @@ void recurse(ITypeDefinition typeDef)
                 }
             }
 
-            ArrayBuilder<IMethodDefinition>? getDeletedMethodDefs(ITypeDefinition typeDef)
+            ArrayBuilder<ITypeDefinitionMember>? getDeletedMemberDefs(ITypeDefinition typeDef)
```

Comment:
Not directly related to this PR: we prefer to avoid captures with local functions (static or just capturing `this` okay) because of allocations and readability. Also consider the same for other local functions here